contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -291,7 +308,7 @@ contract TrueCreditAgency is UpgradeableClaimable {
         return _interest(pool, bucket, borrower);
     }
 
-    function borrow(ITrueFiPool2 pool, uint256 amount) external onlyAllowedBorrowers {
+    function borrow(ITrueFiPool2 pool, uint256 amount) external onlyAllowedBorrowers onlyEligible(pool) {
```

Comment:
Not sure if we need to move it to modifier? Why not put it with other `require`s?

Reply:
agree, the revert message is too context dependant on borrow